frontend/main.go

```diff
@@ -145,16 +146,24 @@ func main() {
 		log.Fatalf("Error creating Kubernetes client: %s", err)
 	}
 
-	// Start server
-	r, err := startHTTPServer(&flags)
-	if err != nil {
-		log.Fatalf("Error starting server: %s", err)
-	}
-
-	log.Println("Starting Odigos UI...")
-	log.Printf("Odigos UI is available at: http://%s:%d", flags.Address, flags.Port)
-	err = r.Run(fmt.Sprintf("%s:%d", flags.Address, flags.Port))
-	if err != nil {
-		log.Fatalf("Error starting server: %s", err)
+	// Attempt to start the server on port 3000 and increment the port number if it's already in use
+	for {
```

Comment:
nit: matter of styling, so completely optional - this for loop has 3 levels of nesting: `for`, `if-else`, `if-else`.

An alternative could be to use just one level of nesting and use `continue` and `break` to structure the code.

So this:
```go
		if err != nil {
			log.Printf("Error starting server on port %d: %s", flags.Port, err)
			flags.Port++ // Try the next port
		} else {
			log.Printf("Starting Odigos UI...")
```

Can become this:
```go
		if err != nil {
			log.Printf("Error starting server on port %d: %s", flags.Port, err)
			flags.Port++ // Try the next port
                         continue
		}

		log.Printf("Starting Odigos UI...")
```